### PR TITLE
Class decorators

### PR DESCRIPTION
**Pull request by levy002**
In this PR, I used the Decorator design pattern to correct the names of the `Person` and its subclasses instances.
- [x] Created the Nameable interface.
- [x] Turned the Person class to Nameable.
- [x] Prepared base Decorator.
- [x] Created CapitalizeDecorator and TrimmerDecorator.

cheers!!

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,5 +1,8 @@
-class Person
+require './nameable'
+
+class Person < nameable
```
Comment (uwadonat) [inline person.rb]: - Let's capitalize `nameable` class here(`Nameable`).

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require './base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name[0...10]
+  end
```
Comment (uwadonat) [inline trimmer_decorator.rb]: - Let this method check  the length of name. The name should have a maximum of 10 characters. If it's longer it should trim the word.

**Review by uwadonat — CHANGES_REQUESTED** [review]
Hi  @levy002 , :hand: 

Good job so far! :mechanical_arm: 
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>
### To highlight! :+1: 
- Your commit and pull request messages are very descriptive ✔️
- Your code is well organized and clean  ✔️
almost



## Status:Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: person.rb (hunk 1 above)
Comment (levy002) [reply person.rb]: Thank you @uwadonat for the excellent suggestions. I've implemented it.

File: trimmer_decorator.rb (hunk 2 above)
Comment (levy002) [reply trimmer_decorator.rb]: Thank you @uwadonat for the excellent suggestions. I've implemented it.

**Review by uwadonat — APPROVED** [review]
## Hi @levy002,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
appr


# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**